### ISSUE TITLE
Issue with WebDriverJS sendKeys

### ISSUE THREAD

**Issue by ColinEberhardt** [angular/angular#6250]
I'm currently working on the TodoMVC Angular 2 implementation, you can see the work in progress via this pull request: https://github.com/tastejs/todomvc/pull/1558

The TodoMVC project uses WebDriverJS in order to automate the testing of each implementation. Unfortunately Angular 2.0 is failing due to some very strange issue. Here's a very simple automated test that illustrates the issue:

```
var webdriver = require('selenium-webdriver');

var driver = new webdriver.Builder().
   withCapabilities(webdriver.Capabilities.chrome()).
   build();

driver.get('http://colineberhardt.github.io/angular2-todo/');
driver.sleep(1000); // sleep to give the app time to bootstrap
var input = driver.findElement(webdriver.By.className('new-todo'));
input.sendKeys('buy some eggs', webdriver.Key.ENTER);
input.sendKeys('do the shopping', webdriver.Key.ENTER);
input.sendKeys('test all things', webdriver.Key.ENTER);
driver.sleep(10000); // sleep so that we can see the results
driver.quit();
```

(The above test is executed against a simplified version of the TodoMVC app, [which can be found here](https://github.com/ColinEberhardt/angular2-todo)).

Unfortunately the text input become completely mangled after the first addition of 'buy some eggs':

![capture](https://cloud.githubusercontent.com/assets/1098110/12092790/2c2ec42c-b2f7-11e5-8dd2-995e74a5635b.PNG)

I have tried all sorts of changes to the TodoMVC app itself, however, the only way I can make the tests work is to add a pause in-between each keystroke:

```
var webdriver = require('selenium-webdriver');

var driver = new webdriver.Builder().
   withCapabilities(webdriver.Capabilities.chrome()).
   build();

driver.get('http://todomvc.com/examples/angular2/');
driver.sleep(1000); // sleep so that we can eliminate startup issues
var input = driver.findElement(webdriver.By.className('new-todo'));

function sendSlowly(text) {
  for (var i = 0, len = text.length; i < len; i++) {
    input.sendKeys(text[i]);
    driver.sleep(1);
  }
  input.sendKeys(webdriver.Key.ENTER);
  driver.sleep(1);
}

sendSlowly('buy some eggs');
sendSlowly('do the shopping');
sendSlowly('test all things');
driver.sleep(10000); // sleep so that we can see the results
driver.quit();
```

This gives the expected result, with three items added to the todo list.

Does anyone have any ideas? I'd love to get the TodoMVC Angular2 app updated!

**Comment by IgorMinar** [comment]
@juliemr can you please take a look?

**Comment by blacksonic** [comment]
I experienced the same issue. Other strange thing when calling ```browser.wait```, it's callback is triggered multiple times.

**Comment by juliemr** [comment]
I am now unable to reproduce this with RC2/RC3. I think the issues has been fixed. Please open a new bug if you can show us a reproducible test case. Thanks!

**Comment by samccone** [comment]
Thanks @juliemr ! Do you have a link back to the commit that fixed this? I am curious as to the origin of the bug :) (if not no worries)

**Comment by juliemr** [comment]
@samccone unfortunately, I don't - if we think it would be worth doing an exhaustive search of versions we could, but all I can say right now is "somewhere between beta0 and RC2".

**Comment by ColinEberhardt** [comment]
Have fun bisecting!

**Comment by samccone** [comment]
ha, it is ok I was just curious because it was a very fascinating issue that I had never seen before.

**Comment by robwormald** [comment]
@samccone it was likely fixed by https://github.com/angular/angular/commit/e5904f40892ca6f53a82d6c5a490d3344a4240fa

tldr - previously we had a weird business where the EventEmitter we use internally (which is an RxJS Subject) was being forced to relay values on the next tick, which caused all kinds of loopy issues with inputs and stuff - see https://github.com/angular/angular/issues/6311 for some similar issues to what this describes.
